packages/react-components/src/components/ActionBar/ActionBar.tsx

```diff
@@ -0,0 +1,132 @@
+import * as React from 'react';
+
+import { ChevronDown } from '@livechat/design-system-icons/react/tabler';
+import cx from 'clsx';
+
+import { ActionMenu, ActionMenuItem } from '../ActionMenu';
+import { Icon } from '../Icon';
+
+import { ActionBarItem, IActionBarOption } from './ActionBarItem';
+
+import styles from './ActionBar.module.scss';
+
+export interface IActionBarProps {
+  /**
+   * The CSS class for menu container
+   */
+  className?: string;
+  /**
+   * The unique id key
+   */
+  id?: string;
+  /**
+   * Array of action bar options
+   */
+  options: IActionBarOption[];
+  /**
+   * Set the key for active element
+   */
+  activeOptionKey?: string | null;
+  /**
+   * Set 'scroll' to disable menu and enable scroll
+   */
+  type?: 'menu' | 'scroll';
+}
+
+const baseClass = 'action-bar';
+
+export const ActionBar: React.FC<IActionBarProps> = ({
+  className,
+  id = 'action-bar-area',
+  type = 'menu',
+  options,
+  activeOptionKey,
+}) => {
+  const [menuItemsKeys, setMenuItemsKeys] = React.useState<string[]>([]);
+  const isScrollType = type === 'scroll';
+  const mergedClassNames = cx(styles[baseClass], className);
+  const observerOptions = {
+    root: document.querySelector(`${id}`),
+    threshold: 1,
+  };
+  const shouldDisplayMenu = !isScrollType && menuItemsKeys.length !== 0;
+
+  const handleIntersection = (entries: IntersectionObserverEntry[]) => {
+    entries.map((entry) => {
+      if (!entry.isIntersecting) {
+        entry.target.setAttribute('tabindex', '-1');
+
+        if (!menuItemsKeys.includes(entry.target.id)) {
```

Comment:
To avoid double-checking the same condition (here and next `if`, in the line 68) we can store this condition in a separate constant or use `if-else` statement.